src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -1,342 +1,235 @@
 package com.mongodb.jdbc;
 
-import java.sql.DatabaseMetaData;
 import java.sql.SQLException;
 import java.sql.Types;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.bson.BsonType;
+
+public enum BsonTypeInfo {
+    BSON_DOUBLE("double", BsonType.DOUBLE, Types.DOUBLE, false, 15, 15, 2, 15, 15, 8),
+    BSON_STRING("string", BsonType.STRING, Types.LONGNVARCHAR, true, 0, 0, 0, null, null, null),
```

Comment:
One of the tests are failing, it's expecting the type to be LONGVARCHAR.
```suggestion
    BSON_STRING("string", BsonType.STRING, Types.LONGVARCHAR, true, 0, 0, 0, null, null, null),
```

Reply:
oh boy great catch, thank you